sycl/doc/extensions/proposed/sycl_ext_oneapi_spirv_queries.asciidoc

```diff
@@ -0,0 +1,205 @@
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 10 specification.
+All references below to the "core SYCL specification" or to section numbers in
+the SYCL specification refer to that revision.
+
+This extension references sections of the
+link:https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html[SPIR-V
+specification] version 1.6, revision 6.
+References below to the "SPIR-V specification" refer to that version.
+
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+
+== Overview
+
+This extension adds new queries that allow an application to identify the
+link:https://www.khronos.org/spirv/[SPIR-V] features that are supported by a
+SYCL device.
+These queries may be useful for applications that use the
+link:../experimental/sycl_ext_oneapi_kernel_compiler_spirv.asciidoc[sycl_ext_oneapi_kernel_compiler_spirv]
+extension to ensure that a SYCL device supports the features required by a
+SPIR-V binary module.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.
+An implementation supporting this extension must predefine the macro
+`SYCL_EXT_ONEAPI_SPIRV_QUERIES`
+to one of the values defined in the table below.
+Applications can test for the existence of this macro to determine if the
+implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== Queries
+
+==== New member functions for the device class
+
+This extension also adds the following member functions to the `device` class,
+which allow the application to query whether the device supports SPIR-V queries,
+and if so, which SPIR-V extended instruction sets, extensions, and capabilities
+the device supports.
+
+|====
+a|
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+class device {
+  bool ext_oneapi_supports_spirv_queries() const;
+};
+----
+!====
+
+_Returns:_ The value `true` if the device supports SPIR-V queries.
+Returns `false` if the device does not support SPIR-V.
+
+a|
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+class device {
+  bool ext_oneapi_supports_spirv_extended_instruction_set(
+    const std::string &name) const;
+};
+----
+!====
+
+_Returns:_ The value `true` if the device supports SPIR-V binary modules that
+import the SPIR-V extended instruction set identified by `name`.
+Returns `false` if the device does not support SPIR-V queries.
```

Comment:
```suggestion
Returns `false` if the device does not support kernels written in SPIR-V.
```

Consistent with my comment above, I'd change this last sentence like this.  Same for the two queries below too.